openedx/core/djangoapps/credentials/tests/mixins.py

```diff
@@ -33,106 +34,92 @@ class CredentialsDataMixin(object):
-                "uuid": "dummy-uuid-6",
-                "certificate_url": "http://credentials.edx.org/credentials/dummy-uuid-6/"
-            }
+            factories.UserCredentials(
+                id=1,
+                username='test',
+                credential=factories.ProgramCredentials(
+                    program_id=1
+                )
+
+            ),
+            factories.UserCredentials(
+                id=2,
+                username='test',
+                credential=factories.ProgramCredentials(
+                    program_id=2
+                )
+            ),
+            factories.UserCredentials(
+                id=3,
+                status='revoked',
+                username='test',
+                credential=factories.ProgramCredentials(
+                    program_id=3
+                )
+            ),
+            factories.UserCredentials(
+                id=4,
+                username='test',
+                credential=factories.CourseCredentials(
+                    certificate_type='honor'
+                )
+            ),
+            factories.UserCredentials(
+                id=5,
+                username='test',
+                credential=factories.CourseCredentials(
+                    course_id='edx/test02/2015'
+                )
+            ),
+            factories.UserCredentials(
+                id=6,
+                username='test',
+                credential=factories.CourseCredentials(
+                    course_id='edx/test02/2015'
+                )
+            ),
         ]
     }
 
     CREDENTIALS_NEXT_API_RESPONSE = {
         "next": None,
         "results": [
-            {
-                "id": 7,
-                "username": "test",
-                "credential": {
-                    "credential_id": 7,
-                    "program_id": 7
-                },
-                "status": "awarded",
-                "uuid": "dummy-uuid-7",
-                "certificate_url": "http://credentials.edx.org/credentials/dummy-uuid-7"
-            },
-            {
-                "id": 8,
-                "username": "test",
-                "credential": {
-                    "credential_id": 8,
-                    "program_id": 8
-                },
-                "status": "awarded",
-                "uuid": "dummy-uuid-8",
-                "certificate_url": "http://credentials.edx.org/credentials/dummy-uuid-8/"
-            }
+            factories.UserCredentials(
+                id=7,
+                username='test',
+                credential=factories.ProgramCredentials(
+                    program_id=7
+                )
+            ),
+            factories.UserCredentials(
+                id=8,
+                username='test',
+                credential=factories.ProgramCredentials(
+                    program_id=8
+                )
+            )
         ]
     }
 
+    def get_program_credentials_data(self):
```

Comment:
The changes you've made here are a step in the right direction. However, the goal is to ultimately get rid of these data mixins so that tests are creating data specific to their needs. Take a look at how [these tests](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/programs/tests/test_backpopulate_program_credentials.py#L62) set up their data. These data mixins are too rigid and removed from the tests, making the tests themselves difficult to write and read. At a minimum, this method should belong to the test class that needs it.